src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/WindowsForms/MyAppDocument.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Xml.Linq;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.WindowsForms;
+
+internal class MyAppDocument
+{
+    private readonly XDocument doc;
+    private readonly XElement root;
+
+    private readonly string name;
+
+    public MyAppDocument(string fileName, string filePath)
```

Comment:
I think you'll need to parse the file each time you get/set a property, rather than once at startup. You cannot assume that external systems won't modify the file over time, and because you keep an instance of `​MyAppDocumen`t​ alive in ​`MyAppFileProcessor`​ (which has the lifetime of the project) any changes made after it's parsed won't be reflected in the UI, for example.

Reply:
Yeah, that's a good point. I added that change in #8252